controllers/AcquisitionSessionController.php

```diff
@@ -0,0 +1,291 @@
+    
+    /**
+     * The name of the worksheet which will be modified
+     */
+    CONST PHIS_SHEET_NAME = "HiddenPhis";
+    
+    /**
+     * The type of the document file  
+     * @var string    
+     */
+    private $documentFileType;
+    
+    /**
+     * The type of the vector required 
+     * @var string 
+     */
+    private $vectorRdfType;
+    
+    /**
+     * This variable store the lasted error 
+     * which has occurred
+     * @var string 
+     */
+    private $error;
+
+    /**
+     * Define the behaviors
+     * @return array
+     */
+    public function behaviors() {
+        return [
+            'verbs' => [
+                'class' => VerbFilter::className(),
+                'actions' => [
+                    'delete' => ['POST'],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Set the right uris for robot field acquisition session metadata
+     * @return Reponse
+     */
+    public function actionGenerateFieldRobot() {
+        $this->vectorRdfType = Yii::$app->params['FieldRobot'];
+        $this->documentFileType = Yii::$app->params['FieldRobotDocument'];
+
+        return $this->generateFile();
+    }
+
+    /**
+     * Set the right uris for uav acquisition session metadata
+     * @return Reponse
+     */
+    public function actionGenerateUav() {
+        $this->vectorRdfType = Yii::$app->params['UAV'];
+        $this->documentFileType = Yii::$app->params['UAVDocument'];
+
+        return $this->generateFile();
+    }
+    
+    /**
+     * Generate the  acquisition session file 
+     * @return Reponse
+     */
+    private function generateFile() {
+        // get the last document and save it on the server
+        $existingFilePath = $this->getTemplateFile();
+        if ($this->error != null) {
+            return $this->error;
+        }
+        // add data retreive from the ws to the worksheet choosen
+        $newFilePath = $this->addHiddenphisSheetData($existingFilePath);
+        if ($this->error != null) {
+            return $this->error;
+        }
+        // save the new file and send it to the user
+        $this->sendTemplateFile($newFilePath);
+        if ($this->error != null) {
+            return $this->error;
+        }
+    }
+
+    /**
+     * Retreive the latest saved document which matches
+     * with the choosen document type
+     * @return mixed null
+     *               or a string with the saved document path
+     */
+    private function getTemplateFile() {
+        // 1. Get installation uri
+        $infrastructureUri = substr(Yii::$app->params['baseURI'], 0, -1);
+        
+        // 2. Get the last acquisition template for the document type required
+        $sessionToken = Yii::$app->session['access_token'];
+        $documentModel = new YiiDocumentModel();
+        $search["documentType"] = $this->documentFileType;
+        $search["concernedItem"] = $infrastructureUri;
+        //SILEX:info
```

Comment:
could you move this tag in the search document for example

Reply:
Done